src/sentry/integrations/migrate.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+from sentry.models import Repository
+from sentry.plugins import plugins
+
+
+class PluginMigrator(object):
+    def __init__(self, integration, organization):
+        self.integration = integration
+        self.organization = organization
+
+    def call(self):
+        for project in self.projects:
+            for plugin in plugins.for_project(project):
+                if self.all_repos_migrated(plugin.slug):
+                    self.disable_for_all_projects(plugin)
+
+    def all_repos_migrated(self, provider):
+        provider = 'visualstudio' if provider == 'vsts' else provider
+
+        return all(
+            r.integration_id is not None
+            for r in self.repos_for_provider(provider)
+        )
+
+    def disable_for_all_projects(self, plugin):
+        for project in self.projects:
+            try:
+                plugin.disable(project=project)
+            except NotImplementedError:
+                pass
+
+    def repos_for_provider(self, provider):
+        return filter(lambda r: r.provider == provider, self.repositories)
+
+    @property
+    def repositories(self):
+        return Repository.objects.filter(
+            organization_id=self.organization.id,
+        )
+
+    @property
+    def projects(self):
+        return self.organization.project_set.all()
```

Comment:
so for this and `def repositories`, i wonder if it'd be better to do something like:

```python
@property
def projects(self):
    if self._projects is None:
        self._projects = list(self.organization.project_set.all())
    return self._projects
```
just to cast to a list/ensure that those queries won't get executed multiple times?  cause otherwise i'm pretty sure they will

Reply:
Can also use `@memoize` + `return list(self.organization.project_set.all())`

https://github.com/getsentry/sentry/blob/master/src/sentry/utils/cache.py#L17

Reply:
yeah that works too, whichever you prefer!

Reply:
Wait, lol, this is my PR... Too many reviews...